[Jiet] iP

hsiaojietng opened the pull request:
Created PR for Duke Project

Commit:
Add Gradle support

hsiaojietng committed:
Implemented Level 1. Greet, Echo, Exit.

hsiaojietng committed:
Implemented DukeList database to act as List with add and list functions

hsiaojietng committed:
Implemented mark as done function and created abstracted methods more

hsiaojietng committed:
Created inherited classes: Todo, Deadline, Events and abstracted the command words into a class "Action"

hsiaojietng committed:
Added testing bat file for semi automated testing

hsiaojietng committed:
Included checks on Action words such as "todo", "deadline", "event" as well as other invalid inputs

hsiaojietng committed:
Added comments on the purpose of these classes: "Deadline", "Event", "Task", "Todo"

hsiaojietng committed:
Implemented Delete function with checks

hsiaojietng committed:
Implemented enums for frequently used command inputs

hsiaojietng committed:
Change name for mark/unmark functions

hsiaojietng committed:
Add FileAction class

hsiaojietng committed:
Change done attribute to '0' or '1'

hsiaojietng committed:
Change presentation of Deadline task

hsiaojietng committed:
Add readTaskFromFile

hsiaojietng committed:
Clean up code

hsiaojietng committed:
Save tasks into text file

hsiaojietng committed:
Fix: save tasks into file

hsiaojietng committed:
Create Date for Deadline & Event tasks

hsiaojietng committed:
Merge branch 'Level-8'

# Conflicts:
#	src/main/java/Action.java
#	src/main/java/Deadline.java
#	src/main/java/Event.java

Fix saving Tasks with LocalDate variable into text file as well as reading Tasks with date information as a String

hsiaojietng committed:
Use OOP and simplify code

hsiaojietng committed:
Fix issues with OOP

hsiaojietng committed:
Create package Duke

hsiaojietng committed:
Add Javadocs and better handle exceptions for unmark,mark & delete commands

hsiaojietng committed:
Update coding standard of app

hsiaojietng committed:
Add findTask function

hsiaojietng committed:
Add JUnit tests for Parser and Ui classes

hsiaojietng committed:
Create some additional exceptions that might be encountered

hsiaojietng commented:
# DukeProMax
> "The opportunities you don't take, will all go to waste either ways" - Some guy on the Internet

DukeProMax allows you to make time for more important things than trying to remember what has to be done.
- Text based
- Easy to learn
- INCREDIBLY FAST

**You just have to,**
1. Download it 
2. Execute the program
3. Write commands as it does your bidding
4. Let it manage all your tasks

Practice Java with the main method

```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

tzhan98 commented on src/duke/Deadline.java:
```diff
@@ -0,0 +1,71 @@
+    /**
+     * Creates duke.Deadline object with provided LocalDate variable
+     */
+    public Deadline(String name, int isDone, LocalDate by) {
```
Should 'isDone' be a boolean? 🤔 
>Boolean variables/methods should be named to sound like booleans.  [source](https://se-education.org/guides/conventions/java/intermediate.html)

tzhan98 commented on src/duke/Deadline.java:
```diff
@@ -0,0 +1,71 @@
+ */
+
+public class Deadline extends Task {
+    protected LocalDate by;
```
Great use of `protected`!

FestiveCat replied:
Maybe it would be better to name this `deadline`, and maybe the class as `DeadlineTask`. What do you think?

tzhan98 commented on src/duke/Event.java:
```diff
@@ -0,0 +1,70 @@
+        super.type = 'E';
+    }
+
+    public void setAt(LocalDate at) {
```
Missing Documentation

tzhan98 commented on src/duke/Parser.java:
```diff
@@ -0,0 +1,337 @@
+public class Parser {
+    protected String userInput;
+    protected TaskList tasks;
+    protected String[] listOfUserInput;
```
Not sure if it should be `listOfUserUnputs` instead of `listOfUserInput`
>Plural form should be used on names representing a collection of objects.

tzhan98 commented on src/duke/Parser.java:
```diff
@@ -0,0 +1,337 @@
+    public void checkTaskNumber() throws DukeException {
+        int taskNumber = obtainTaskNumber();
+        if (taskNumber >= tasks.getTaskListSize() || taskNumber < 1) {
+            throw new DukeException("Please provide the correct Task number! Maybe review the list of tasks first,\nAnd then execute the command for mark/unmark/delete!\n");
```
Too long, line wrapping should be used.
>Line length should be no longer than 120 chars.

tzhan98 commented on src/duke/Storage.java:
```diff
@@ -0,0 +1,138 @@
+     * places duke.Task objects into the current DukeList
+     */
+    public void readTasksFromFile() throws FileNotFoundException {
+        ArrayList<Task> readTasks = new ArrayList<>();
```
Should `readTasks` be a noun? Something like `tasksThatHaveBeenRead`

FestiveCat replied:
In this case, read can mean the past-tense version of it, like "I read a book last week", and not "I read textbooks for fun"

hsiaojietng replied:
Ah you're right, this can be quite confusing for people who do not know how I code. Thanks a lot!

tzhan98 replied:
I see, sorry for my bad English haha 🤣

hsiaojietng replied:
No no, I am saying that the way I named the function wasn't informational enough for users (confusing to them), that's why thanks for the recommendation! LOL

tzhan98 reviewed (commented):
LGTM. 
- Liked how you made use of `Enums`
- Great that you kept to the coding standards
- Just a few minor mistakes/suggestions

Keep up the good work! 👍 😄

FestiveCat commented on src/duke/Deadline.java:
```diff
@@ -0,0 +1,71 @@
+    /**
+     * Setting duke.Deadline object's date it should end by
+     */
+    public void setStringToLocalDate(String date) {
```
What's your rationale for naming this method as `setString...`? Do you think it is hard to tell what "String" the caller of the function is setting?

FestiveCat commented on src/duke/DukeException.java:
```diff
@@ -0,0 +1,21 @@
+ * @version 1.0
+ * @since   2022-1-15
+ */
+public class DukeException extends Exception {
```
Hmm, what's a Duke Exception? When is this exception thrown?

FestiveCat commented on src/duke/Parser.java:
```diff
@@ -0,0 +1,337 @@
+    String line = "\n____________________________________________________________\n";
+
+    enum Commands {
+        todo, deadline, event, list, mark, unmark, delete, find, bye
```
Shouldn't the various states of the enumerations should be in SNAKE_CASE because they are constants? ie:
```ruby
Commands.TODO
Commands.DEADLINE
...
```

FestiveCat commented on src/duke/Parser.java:
```diff
@@ -0,0 +1,337 @@
+    /**
+     * Obtains title. Applicable for creating Todo, duke.Deadline and duke.Event
+     */
+    public String obtainTitle() throws DukeException {
```
I think it will help greatly if this method was named more descriptively, in particular, what is a `Title`? What do you think?

FestiveCat commented on src/duke/Parser.java:
```diff
@@ -0,0 +1,337 @@
+    }
+
+    /**
+     * Makes a call on duke.TaskList's unmark()
```
I'm not too sure if this is descriptive enough. Let's say if this was another method and not an unmark task method, and the Javadocs said something like: `Makes a call on foo's bar()`. There isn't much description here, don't you think?

FestiveCat commented on src/duke/Parser.java:
```diff
@@ -0,0 +1,337 @@
+    protected TaskList tasks;
+    protected String[] listOfUserInput;
+    protected Storage storage;
+    String line = "\n____________________________________________________________\n";
```
Honestly found this funny and gave me a chuckle because of how apt this is. Good job!

hsiaojietng replied:
HAHAHA this is the best comment I've ever seen in my entire life

FestiveCat commented on src/duke/Storage.java:
```diff
@@ -0,0 +1,138 @@
+    protected String filepath;
+    protected FileWriter fileWriter;
+    protected FileReader fileReader;
+    protected BufferedWriter writer;
```
Do you think it's better to name the BufferedWriter/Readers as buffWriter or buffReader, since there are similar fields for FileWriter/Reader?

FestiveCat commented on src/duke/Ui.java:
```diff
@@ -0,0 +1,61 @@
+
+    /** Returns a pre-made command message to users  */
+    public String askForCommand() {
+        StringBuilder commandLine = new StringBuilder();
```
commandLine can be a little misleading, don't you think? As in, Command Line Arguments or Command Line Interface

FestiveCat commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,33 @@
+    TaskList tasks = new TaskList();
+    Storage storage = new Storage("data/duke.txt", tasks);
+    @Test
+    public void checkTodo() throws DukeException {
```
It might be better to name your test methods in the format of: `<methodName>Test_<inputDescription>_<expectedOutcome>`.
For example, `parserConstructorTest_toDoTask_success`
What do you think?

hsiaojietng replied:
Wow interesting, I've never known test is supposed to be written like that. But it makes sense and makes it more descriptive. Thanks a lot!

FestiveCat reviewed (commented):
I think there are quite a few kinks to iron out, especially on better naming of variables/methods. Good job in implementing the logic though!

hsiaojietng commented:
Overall, with your review, I think I learned a lot and made so many sensible changes to my code. Cheers!

hsiaojietng committed:
Fix nits from PR review

hsiaojietng committed:
Merge branch 'master' into add-gradle-support

# Conflicts:
#	build.gradle
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties
#	gradlew
#	gradlew.bat

hsiaojietng committed:
Change such that project becomes Gradle supported

hsiaojietng committed:
Trial push

hsiaojietng committed:
Revert "Trial push"

This reverts commit e651c4c9e5813f35b5e5b0524182951ff155c830.

hsiaojietng committed:
Set up JavaFX for Duke

hsiaojietng committed:
Revert "Set up JavaFX for Duke"

This reverts commit 7c431c1479e16281a09283f2bbd5b7a434c02925.

hsiaojietng committed:
Simplify Parser

hsiaojietng committed:
Add JavaFX

hsiaojietng committed:
Fix GUI issues

hsiaojietng committed:
Simplify code for deleteTask

hsiaojietng committed:
Add additional JavaDocs

hsiaojietng committed:
MainWindow Class: create a class for MainWindow

Require a Java class to handle the logic of main GUI of Duke application

This will help to handle the input of users and based on the logic of Duke, return the respective response in the GUI.

Let's think of a way to handle the change between the Ui of the dialog between user and Duke and how the response is produced from the user input.

By separating this logic away from the Duke class, we can reduce the complexity of the code in the Duke class

Relevant Info: https://se-education.org/guides/tutorials/javaFxPart4.html

hsiaojietng committed:
Fix checkstyle for Deadline

hsiaojietng committed:
Fix Java version for JavaFX

hsiaojietng committed:
Assertions: Calling Assertions to make checks to assumptions

Some codes require us to make certain assumptions for it to work as intended.

Creating Assertions to make sure that all these assumptions are always true reduces the amount of bugs that can pop up.

Let's create some Assertions in relevant classes to reduce failure of assumptions.

Use junit asserttrue to check if a particular process is true.

Refer to this website for additional information on how to use assertTrue: https://www.baeldung.com/java-file-directory-exists

hsiaojietng committed:
Add sort function on Task title

hsiaojietng committed:
Check code quality consistently

hsiaojietng committed:
Change property of Jar file

hsiaojietng committed:
Set up readme

hsiaojietng committed:
Set theme jekyll-theme-slate

hsiaojietng committed:
Add Ui.png

hsiaojietng committed:
Update README.md

hsiaojietng committed:
Updated readme again

hsiaojietng committed:
Merge branch 'master' of https://github.com/hsiaojietng/ip

hsiaojietng committed:
Create PieChart for number of Tasks based on type

hsiaojietng committed:
Merge pull request #2 from hsiaojietng/Learn-Tabpane

Learn and create Pie Chart

hsiaojietng committed:
Create Statistic.fxml file

hsiaojietng committed:
Merge branch 'Learn-Piechart'

# Conflicts:
#	src/main/resources/duke/view/Statistic.fxml

hsiaojietng committed:
Merge branch 'master' of https://github.com/hsiaojietng/ip

hsiaojietng committed:
Give credit to code reused and create Ui.png

hsiaojietng committed:
Remove assert for shadowJar to work

hsiaojietng committed:
Set label for dialogs scalable

hsiaojietng committed:
Change Duke-tator readme

hsiaojietng committed:
Add creation of data folder and duke.txt file function

hsiaojietng committed:
Change readme and updated unit tests

hsiaojietng committed:
Change Ui picture of Duke-tator

hsiaojietng committed:
Add javadocs

hsiaojietng committed:
Improve code quality on Brace style, Dead code and Header comments
